src/test/java/org/alfresco/repo/dictionary/DiffModelTest.java

```diff
@@ -491,6 +491,64 @@ public void testNonIncUpdatePropertiesRemovedFromMandatoryAspect()
         assertEquals(1, countDiffs(modelDiffs, M2ModelDiff.TYPE_PROPERTY, M2ModelDiff.DIFF_DELETED));
     }
     
+    /**
+     * Changing a property from mandatory/enforced/protected to NON mandatory/enforced/protected
+     * is an incremental change and it should be allowed.
+     */
+    public void testIncChangeMandatoryProperties()
+    {
+        ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(AbstractModelTest.MODEL8_XML.getBytes());
+        M2Model model = M2Model.createModel(byteArrayInputStream);
+        QName modelName = dictionaryDAO.putModel(model);
+        CompiledModel previousVersion = dictionaryDAO.getCompiledModel(modelName);
+        
+        byteArrayInputStream = new ByteArrayInputStream(AbstractModelTest.MODEL8_CHANGE_MANDATORY_PROPERTIES_ASPECTS_XML.getBytes());
+        model = M2Model.createModel(byteArrayInputStream);
+        modelName = dictionaryDAO.putModel(model);
+        CompiledModel newVersion = dictionaryDAO.getCompiledModel(modelName);
+        
+        List<M2ModelDiff> modelDiffs = dictionaryDAO.diffModel(previousVersion, newVersion);
+        
+        for (M2ModelDiff modelDiff : modelDiffs)
+        {
+            System.out.println(modelDiff.toString());
+        }   
+        
+        assertEquals(2, modelDiffs.size());
+        
+        assertEquals(1, countDiffs(modelDiffs, M2ModelDiff.TYPE_ASPECT, M2ModelDiff.DIFF_UNCHANGED));
+        assertEquals(1, countDiffs(modelDiffs, M2ModelDiff.TYPE_PROPERTY, M2ModelDiff.DIFF_UPDATED_INC));
```

Comment:
Rather than these three assertions - I think we should just use a single Set comparison (possibly with an assertion message).

Reply:
Same comment for the other test.